frontend/src/components/set-org/SetOrg.jsx

```diff
@@ -23,23 +25,29 @@ function SetOrg() {
   const handleException = useExceptionHandler();
 
   useEffect(() => {
-    if (state === null || signedInOrgId) {
-      navigate("/");
-    }
-  }, []);
+    const fetchData = async () => {
+      try {
+        const userSessionData = await userSession();
+        const signedInOrgId = userSessionData?.organization_id;
+        if (state === null || (signedInOrgId && signedInOrgId !== "public")) {
+          navigate("/");
+          return;
```

Comment:
@muhammad-ali-e I think it should be safe to remove the `return` statement, as the logic concludes at this point.

Reply:
updated